File: Data_Structures/Hashing/SeperateChaining.java
import java.util.Scanner;

class SeperateChaining {

    public static void main(String[] args) {
        final int SIZE = 10;
        Scanner scn = new Scanner(System.in);

        Node[] hash_table = new Node[SIZE];
        for(int i=0; i<SIZE; i++){
            hash_table[i] = null;
        }
        // Used to clear the screen
        System.out.print("\033[H\033[2J");  
        System.out.flush();  
        while(true){
            for(int i=0; i<SIZE; i++){
                Node current = hash_table[i];
                System.out.print("[ " + i + " ]--> ");
                if(current!=null){
                    current.print_chain();
                } 
                System.out.println();
            }

            System.out.print("Enter new node : ");
            String given = scn.nextLine();
            if("exit".equals(given)) break;
            int input = Integer.parseInt(given);
            Node newNode = new Node(input);
            if(hash_table[input%10] == null){
                hash_table[input%10] = newNode;
            } 
            else{
                Node last = hash_table[input%10];
                while(last.next != null) {
                    last = last.next;
                }
                last.next = newNode;
            }
        }
        scn.close();
    }
}

File: Data_Structures/Hashing/SeperateChaining.cpp
#include<bits/stdc++.h>
#define SIZE 10

class Node {
    public:
        int number;
        Node *next;

        void print_chain(){
            printf("( %d )",this->number);
            Node *temp = next;
            while(temp!=NULL){
                printf("( %d )",temp->number);
                temp = temp->next;
            }
        }
};

void insert_node_to_chain(int value, Node **head){
    Node *ptr = (*head);
    Node *new_node = (Node*)malloc(sizeof(Node));
    new_node->number = value;
    new_node->next = NULL;
    if(*head == NULL){
        *head = new_node;
    }
    else{
        while(ptr->next !=NULL){
            ptr = ptr->next;
        }
        ptr->next = new_node;
    }
}

int main(){
    Node *hash_table[SIZE];
    for(int i=0; i<SIZE; i++){
        hash_table[i]=NULL;
    }
    while(1){
        system("clear");
        for(int i=0; i<SIZE; i++){
            Node *ptr = hash_table[i];
            printf("[%d]--> ",i);
            if(ptr!=NULL) ptr->print_chain();
            printf("\n");
        }
        std::cout << "Enter new node: ";
        int input;
        scanf("%d",&input);
        insert_node_to_chain(input, &hash_table[input%10]);
    }
    return 0;
}
